include/certifier_framework.h

```diff
@@ -109,40 +146,155 @@ namespace certifier {
     bool Attest(const string& enclave_type,
-      int what_to_say_size, byte* what_to_say,
-      int* size_out, byte* out);
+                int what_to_say_size, byte* what_to_say,
+                int* size_out, byte* out);
 
-    // Protect Support
+    // -------------------------------------------------------------------
+    // Protect Support: Wrappers around platform-specific Seal(), Unseal()
     // -------------------------------------------------------------------
 
+    // ------------------------------------------------------------------------
+    // Protect unencrypted data using user-specified 'key'. This uses the
+    // platform-specific Seal() interface.
+    //
+    // Parameters:
+    // - enclave_type - String describing the enclave type. (See above)
+    // - key - User-defined 'key'
+    // - size_unencrypted_data - Length of unencrypted (input) data
+    // - unencrypted_data - Unencrypted (input) data
+    // - size_protected_blob - (Out) Length of encrypted data, 'blob'
+    // - blob - (Out) Encrypted data
+    //
+    // Returns: True on a successful operation. False, otherwise
+    // ------------------------------------------------------------------------
     bool Protect_Blob(const string& enclave_type,
-      key_message& key, int size_unencrypted_data, byte* unencrypted_data,
-      int* size_protected_blob, byte* blob);
+                      key_message& key, int size_unencrypted_data,
+                      byte* unencrypted_data, int* size_protected_blob,
+                      byte* blob);
+
+    // ------------------------------------------------------------------------
+    // Unprotect the encrypted data. This uses the platform-specific Unseal()
+    // interface.
+    //
+    // Parameters:
+    // - enclave_type - String describing the enclave type. (See above)
+    // - size_protected_blob, - Length of encrypted (input) data
+    // - protected_blob - Encrypted (input) data
+    // - key - (Out) User-defined 'key' that was used for protection
+    // - size_of_unencrypted_data - (Out) Length of unencrypted data, 'data'
+    // - data - (Out) Unencrypted data
+    //
+    // Returns: True on a successful operation. False, otherwise
+    // ------------------------------------------------------------------------
     bool Unprotect_Blob(const string& enclave_type,
-      int size_protected_blob, byte* protected_blob,
-      key_message* key, int* size_of_unencrypted_data, byte* data);
+                        int size_protected_blob, byte* protected_blob,
+                        key_message* key, int* size_of_unencrypted_data,
+                        byte* data);
+
+    // ------------------------------------------------------------------------
+    // Reprotect an encrypted data with an internally-generated new key that
+    // is used for re-protecting the data. (RESOLVE: How does this business of
+    // cipher_key_byte_size() in implementation work out.)
```

Comment:
This interface's description needs a lot of help. Let's walk thru RESOLVEs I've left behind ... and straighten this out.